src/psyclone/psyir/backend/fortran.py

```diff
@@ -1629,3 +1642,20 @@ def call_node(self, node):
             return f"{self._nindent}call {node.routine.name}({args})\n"
         # Otherwise it is inside-expression function call
         return f"{node.routine.name}({args})"
+
+    def kernelfunctor_node(self, node):
+        '''
+        Translate the Kernel functor into Fortran.
+
+        :param node: the PSyIR node to translate.
+        :type node: :py:class:`psyclone.domain.common.algorithm.KernelFunctor`
+
+        :returns: the equivalent Fortran code.
+        :rtype: str
+
+        '''
+        result_list = []
+        for child in node.children:
+            result_list.append(self._visit(child))
+        args = ", ".join(result_list)
+        return f"{node.name}({args})"
```

Comment:
Not your fault, but line 358: super() without arguments

Reply:
Fixed.